### PR TITLE
Add mypy to pre-commit

### PR DESCRIPTION
**Pull request by illusional**
Feel free to close if not desired, but we find it handy internally to use mypy as it's better at catching type errors.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,3 +16,17 @@ repos:
     rev: 6.0.0
     hooks:
       - id: pydocstyle
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.961
```
Comment (ch-kr) [inline .pre-commit-config.yaml:21]: why not use the most recent version (1.0.0)?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -16,3 +16,17 @@ repos:
...
+      - id: mypy
+        args:
+          [
+            --pretty,
+            --show-error-codes,
+            --no-strict-optional,
+            --ignore-missing-imports,
+            --install-types,
+            --non-interactive,
+          ]
```
Comment (ch-kr) [inline .pre-commit-config.yaml:32]: I'm going to sneak this into this PR as well (another item I've been meaning to add):
```suggestion
          ]
    - repo: https://github.com/pycqa/isort
      rev: 5.10.1 # This should be kept in sync with the version in requirements-dev.in
      hooks:
        - id: isort
          args: ["--profile", "black", "--filter-files"]
          files: rmc
```

**Review by ch-kr — COMMENTED** [review]
thanks for adding this -- this has actually been on my todo list for some time!

**Comment by ch-kr** [comment]
basic question, wouldn't we also need to add `mypy` to `requirements-dev.in`? (and `isort`)

**Comment by illusional** [comment]
The `pre-commit` library will keep everything installed and correct version itself (except pylint which I listed as `local`, because it works better with packages that way), so no need to add them explicitly to requirements-dev.

**Comment by illusional** [comment]
(Just noting that mypy / isort aren't being run in CI without merging #268 first.

**Comment by ch-kr** [comment]
Yes, I just merged that other PR. Thank you for adding it, and for adding isort! My only remainining question here is around the version; it seems like the most recent `mypy` version is 1.0.0. Is there a reason you prefer v0.961?

**Comment by ch-kr** [comment]
pre-commit failures fixed in https://github.com/broadinstitute/regional_missense_constraint/pull/275

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -16,6 +16,25 @@ repos:
     rev: 6.0.0
     hooks:
       - id: pydocstyle
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.961
```
Comment (ch-kr) [inline .pre-commit-config.yaml:20]:
```suggestion
    rev: v1.0.1
```

**Comment by ch-kr** [comment]
closing; duplicated in https://github.com/broadinstitute/regional_missense_constraint/pull/278

**Pull request closed without merging**